InteractiveHtmlBom/ecad/fusion_eagle.py

```diff
@@ -0,0 +1,618 @@
+                    'thickness': size * ratio,
+                    'attr': [] if not mirrored else ['mirrored'],
+                    'angle': angle
+                }
+
+                self.font_parser.parse_font_for_string(text)
+                if mirrored:
+                    self.pcbdata['drawings']['silkscreen']['B'].append(dwg)
+                else:
+                    self.pcbdata['drawings']['silkscreen']['F'].append(dwg)
+
+    @staticmethod
+    def _segments_to_polygon(segs):
+        polygon = []
+        for vertex in segs.iter('vertex'):
+            x, y = float(vertex.attrib['x']), -float(vertex.attrib['y'])
+            polygon.append([x, y])
+        return polygon
+
+    def _add_zone(self, poly, net):
+        dest = self.pcbdata['zones']['F'] if poly.attrib['layer'] == \
+                                             self.TOP_COPPER_LAYER \
+            else self.pcbdata['zones']['B']
+
+        if poly.tag == 'polygonpour':
+            segs = poly.find('polygonfilldetails').find('polygonshape') \
+                .find('polygonoutlinesegments')
+        else:
+            segs = poly
+
+        zone = {'net': net,
+                'polygons': []}
+        zone['polygons'].append(self._segments_to_polygon(segs))
+        dest.append(zone)
+
+    def _add_parsed_font_data(self):
+        for (c, wl) in self.font_parser.get_parsed_font().items():
+            if c not in self.pcbdata['font_data']:
+                self.pcbdata['font_data'][c] = wl
+
+    def parse(self):
+        ext = os.path.splitext(self.file_name)[1]
+
+        if ext.lower() == '.fbrd':
+            with zipfile.ZipFile(self.file_name) as myzip:
+                brdfilename = [fname for fname in myzip.namelist() if
+                               os.path.splitext(fname)[1] == '.brd']
+                with myzip.open(brdfilename[0]) as brdfile:
+                    return self._parse(brdfile)
+
+        elif ext.lower() == '.brd':
+            with io.open(self.file_name, 'r', encoding='utf-8') as brdfile:
+                return self._parse(brdfile)
+
+    def _parse(self, brdfile):
+        try:
+            brdxml = ElementTree.parse(brdfile)
+        except:
+            self.logger.error("Exception occurred trying to parse {0}"
+                              .format(brdfile.name))
+            return None, None
+        if brdxml is None:
+            self.logger.error("No data was able to be parsed from {0}"
+                              .format(brdfile.name))
+            return None, None
+
+        # Pick out key sections
+        root = brdxml.getroot()
+        board = root.find('drawing').find('board')
+        plain = board.find('plain')
+        elements = board.find('elements')
+        signals = board.find('signals')
+
+        # Determine minimum via annular ring from board design rules
+        # (Needed in order to calculate through-hole pad diameters correctly)
+        mv = [el.attrib['value'] for el in root.iter('param') if
```

Comment:
Is this guaranteed to have `rlMinViaOuter` parameter? This will throw if eagle omits unchanged default values by chance.

Reply:
Adding logic w/ warnings